### PR TITLE
Done bai2

### PR DESCRIPTION
**Pull request by quixun**

### CODE REVIEW

File: Bai2/Bai2_2.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  - Example: 
+    + Input: arr1 = [1, 2, 3, 4], arr2 = [3, 4, 5, 6]
+    + Output: [3, 4]
+*/
+
+const filterArr = (arr1, arr2) => {
+  const newArr = [];
+  const arr1Length = arr1.length;
+  const arr2Length = arr2.length;
+  for (let i = 0; i < arr1Length; i++) {
```
Comment (nguyenvanman) [inline Bai2/Bai2_2.js]: array.filter

**Review by nguyenvanman — CHANGES_REQUESTED** [review]

File: Bai2/Bai2_2.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  - Example: 
+    + Input: arr1 = [1, 2, 3, 4], arr2 = [3, 4, 5, 6]
+    + Output: [3, 4]
+*/
+
+const filterArr = (arr1, arr2) => {
+  return arr1.filter((item) => arr2.includes(item));
+};
+
+console.log(filterArr([1, 2, 3, 4], [3, 4, 5, 6]));
```
Comment (nguyenvanman) [inline Bai2/Bai2_2.js]: remove console.log

File: Bai2/Bai2_2.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/* 
+    2. Write a function that takes two arrays and returns a new array containing the elements that are present in both arrays.
+  - Example: 
+    + Input: arr1 = [1, 2, 3, 4], arr2 = [3, 4, 5, 6]
+    + Output: [3, 4]
+*/
+
+const filterArr = (arr1, arr2) => {
```
Comment (nguyenvanman) [inline Bai2/Bai2_2.js]:
```suggestion
export const filterArr = (arr1, arr2) => {
```

File: Bai2/Bai2_2.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  return arr1.filter((item) => arr2.includes(item));
+};
+
+console.log(filterArr([1, 2, 3, 4], [3, 4, 5, 6]));
+
+module.exports = filterArr;
```
Comment (nguyenvanman) [inline Bai2/Bai2_2.js]: can be removed

File: Bai2/.gitignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+/node_modules
+*.json
```
Comment (nguyenvanman) [inline Bai2/.gitignore]: why do you need to ignore the json file?

File: Bai2/package.json [hunk 6]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (nguyenvanman) [inline Bai2/package.json]: add this to the root level of prj, to be able to use in the whole project

File: Bai2/filterArr.test.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+const filterArr = require("./Bai2_2.js");
+
+test("compare arr1 and arr2 return the elements that are present in both arrays", () => {
```
Comment (nguyenvanman) [inline Bai2/filterArr.test.js]: add one more test for the case that 2 arrs have no common values

**Review by nguyenvanman — CHANGES_REQUESTED** [review]

File: Bai2/package.json (hunk 6 above)
Comment (nguyenvanman) [reply Bai2/package.json]: separator this into another PR

**Review by nguyenvanman — CHANGES_REQUESTED** [review]

File: Bai2/.gitignore (hunk 5 above)
Comment (quixun) [reply Bai2/.gitignore]: i don't wanna push folder node_modules to github because it's so heavy

Comment (nguyenvanman) [reply Bai2/.gitignore]: No, this *.json

Comment (quixun) [reply Bai2/.gitignore]: because I got warnings
warning: in the working copy of 'package-lock.json', LF will be replaced by CRLF the next time Git touches it
warning: in the working copy of 'package.json', LF will be replaced by CRLF the next time Git touches it
therefore i find it unnecessary to push

File: Bai2/bai2_2.test.js [hunk 8]
```diff
@@ -0,0 +1,4 @@
+const filterArr = require("./bai2_2");
+test("compare arr1 and arr2 return the elements that are present in both arrays", () => {
```
Comment (nguyenvanman) [inline Bai2/bai2_2.test.js]: add more test for the case empty arrays

**Review by nguyenvanman — CHANGES_REQUESTED** [review]

File: Bai2/bai2_2.test.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
...
```
Comment (nguyenvanman) [inline Bai2/bai2_2.test.js]: The description should indicate the specific function or method being tested and what aspect of it is under examination

**Review by nguyenvanman — CHANGES_REQUESTED** [review]

File: Bai2/bai2_2.test.js [hunk 10]
```diff
@@ -0,0 +1,9 @@
+const filterArr = require("./bai2_2");
+
+test("filterArr should return a new array including the elements that are present in both arrays", () => {
+  expect(filterArr([1, 2, 3, 4], [3, 4, 5, 6])).toStrictEqual([3, 4]);
+});
+
+test("filterArr should return the null array when there is not any element present in both arrays", () => {
```
Comment (nguyenvanman) [inline Bai2/bai2_2.test.js]: It's not null
```suggestion
test("filterArr should return empty array if two arrays have no common element", () => {
```

**Review by nguyenvanman — CHANGES_REQUESTED** [review]

**Review by nguyenvanman — APPROVED** [review]
